script.js

```diff
@@ -0,0 +1,96 @@
+const userInput = document.querySelector(".userInput");
+const itemsList = document.querySelector(".itemsList");
+let itemsCounter = 0;
+
+const createNewItem = (item) => {
+  const newItem = document.createElement("li");
+  newItem.dataset.item = itemsCounter;
+  newItem.dataset.itemName = item;
+  itemsList.appendChild(newItem);
+  addItemsFeatures(newItem);
+};
+
+const addItemsFeatures = (newItem) => {
+  const itemName = document.createElement("span");
+  itemName.innerHTML = newItem.dataset.itemName;
+  const quantity = document.createElement("input");
+  quantity.type = "number";
+  quantity.placeholder = "Qty";
+  const comments = document.createElement("input");
+  comments.placeholder = "Comments";
+  const check = document.createElement("input");
+  check.type = "checkbox";
+  check.dataset.checkbox = itemsCounter;
+  const remove = document.createElement("button");
+  remove.innerHTML = "REMOVE";
+  remove.dataset.remove = itemsCounter;
+  newItem.append(itemName, quantity, comments, check, remove);
+};
+
+userInput.addEventListener("keydown", (key) => {
+  if (key.keyCode === 13) {
+    const item = userInput.value.replace(/^\w/, (c) => c.toUpperCase());
+    const VALID_ITEM = /^[^0-9]{2,}$/;
+    if (VALID_ITEM.test(item)) {
+      itemsCounter++;
+      createNewItem(item);
+      userInput.value = "";
+      userInput.focus();
+    }
+  }
+});
+
+document.querySelector(".insertItemBtn").addEventListener("click", function () {
+  const item = userInput.value.replace(/^\w/, (c) => c.toUpperCase());
+  const VALID_ITEM = /^[^0-9]{2,}$/;
+  if (VALID_ITEM.test(item)) {
+    itemsCounter++;
+    createNewItem(item);
+    userInput.value = "";
+    userInput.focus();
+  }
+});
+
+document.querySelector(".orderAZ").addEventListener("click", function () {
```

Comment:
prefer the es6 syntax (arrow function rather than anonymous function). [you can get the element from the event.target]